src/Microsoft.DotNet.Build.Tasks/GenerateCurrentVersion.cs

```diff
@@ -93,15 +94,26 @@ public override bool Execute()
             return true;
         }
 
-        public string GetCurrentRevisionForBuildName(string buildId)
+        public bool SetVersionAndRevisionFromBuildId(string buildId)
         {
-            Regex regex = new Regex(@"\.(\d+)$");
+            Regex regex = new Regex(@"(\d{8})\.(\d+)$");
+            string dateFormat = "yyyyMMdd";
             Match match = regex.Match(buildId);
-            if (match.Success && match.Groups.Count > 1)
+            if (match.Success && match.Groups.Count > 2)
             {
-                return match.Groups[1].Value;
+                DateTime buildIdDate;
+                if (!DateTime.TryParseExact(match.Groups[1].Value, dateFormat, enUS, DateTimeStyles.AssumeLocal, out buildIdDate))
```

Comment:
AssumeLocal here is just telling the parsing to assume the string it is parsing is matching the local timezone which may or may not be correct in all cases but for now I think we are safe to assume that.